### PR TITLE
proto_format: Shift `proto_sync` to bazel

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:

This doesnt resolve the performance issues with proto_format/sync but is the first step, bringing these into bazel will allow performance to be addressed subsequently.

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
succesfully failing proto_sync run https://dev.azure.com/cncf/envoy/_build/results?buildId=110824&view=logs&j=55bc2ab9-36c1-5ab8-58a6-8c258728eeff&t=33b57bf9-ef13-58c8-bc2b-f804aeb16d39&l=1505

File: tools/proto_format/BUILD [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+)
+
+envoy_genjson(
+    name = "data_srcs",
+    srcs = [
+        ":proto_targets",
+        "@envoy_api//bazel:external_proto_deps",
+    ],
+    filter = """
+    {external_proto_deps: .[1],
```
Comment (phlax) [inline tools/proto_format/BUILD:47]: <>

**Comment by phlax** [comment]
better failing run with no other errors https://dev.azure.com/cncf/envoy/_build/results?buildId=110832&view=logs&j=55bc2ab9-36c1-5ab8-58a6-8c258728eeff&t=33b57bf9-ef13-58c8-bc2b-f804aeb16d39&l=589

**Comment by phlax** [comment]
needs descriptor path

**Review by keith — APPROVED** [review]

**Review by htuch — APPROVED** [review]

**Pull request merged**